pennylane/templates/tensornetworks/ttn.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the TTN template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+import warnings
+import pennylane as qml
+import pennylane.numpy as np
+from pennylane.operation import Operation, AnyWires
+
+
+def compute_indices(wires, n_block_wires):
+    """Generate a list containing the wires for each block.
+
+    Args:
+        wires (Iterable): wires that the template acts on
+        n_block_wires (int): number of wires per block
+
+    Returns:
+        layers (array): array of wire indices or wire labels for each block
+    """
+
+    n_wires = len(wires)
+
+    if n_block_wires % 2 != 0:
+        raise ValueError(f"n_block_wires must be an even integer; got {n_block_wires}")
+
+    if n_block_wires < 2:
+        raise ValueError(
+            f"number of wires in each block must be larger than or equal to 2; got n_block_wires = {n_block_wires}"
+        )
+
+    if n_block_wires > n_wires:
+        raise ValueError(
+            f"n_block_wires must be smaller than or equal to the number of wires; "
+            f"got n_block_wires = {n_block_wires} and number of wires = {n_wires}"
+        )
+
+    if not np.log2(n_wires / n_block_wires).is_integer():
```

Comment:
Could this ever throw an error due to numerical precision? I guess not if `n_wires` and `n_block_wires` are integers, but good to check

Reply:
We can get a false `True` when n_block_wires is e.g. `4.000000000000001`.
I think users will feed integers to the template. This should work both when the arguments are of type `float` or of type `int`.
I could raise an error or warning when `n_wires` or `n_block_wires` are not integers.

Reply:
No it's fine as is, thanks!